src/stirling/obj_tools/elf_reader.cc

```diff
@@ -167,7 +167,7 @@ StatusOr<std::unique_ptr<ElfReader>> ElfReader::Create(
 
   elf_reader->binary_path_ = binary_path;
 
-  if (!elf_reader->elf_reader_.load(binary_path, /* skip_segments */ true)) {
+  if (!elf_reader->elf_reader_.load(binary_path, /* skip_segments */ false)) {
```

Comment:
Were the segments skipped intentionally or was this not a need before?

Reply:
May have been unintentional. Can you say more about what you're looking for in he segments?

Reply:
Since segments are what end up memory mapped, we need to find the segment offset where the `.text` section is located to find the function's memory address.

Here is an example `libcrypto.so.1.1` file and the corresponding segments:

```
vagrant@vagrant:/vagrant$ readelf -lW libcrypto.so.1.1

Elf file type is DYN (Shared object file)
Entry point 0x79000
There are 9 program headers, starting at offset 64

Program Headers:
  Type           Offset   VirtAddr           PhysAddr           FileSiz  MemSiz   Flg Align
  LOAD           0x000000 0x0000000000000000 0x0000000000000000 0x077930 0x077930 R   0x1000
  LOAD           0x078000 0x0000000000078000 0x0000000000078000 0x19a80d 0x19a80d R E 0x1000
  LOAD           0x213000 0x0000000000213000 0x0000000000213000 0x090420 0x090420 R   0x1000
  LOAD           0x2a3ab0 0x00000000002a4ab0 0x00000000002a4ab0 0x02d0c0 0x0314f0 RW  0x1000
  DYNAMIC        0x2ce4c8 0x00000000002cf4c8 0x00000000002cf4c8 0x000230 0x000230 RW  0x8
  NOTE           0x000238 0x0000000000000238 0x0000000000000238 0x000024 0x000024 R   0x4
  GNU_EH_FRAME   0x256c58 0x0000000000256c58 0x0000000000256c58 0x00c904 0x00c904 R   0x4
  GNU_STACK      0x000000 0x0000000000000000 0x0000000000000000 0x000000 0x000000 RW  0x10
  GNU_RELRO      0x2a3ab0 0x00000000002a4ab0 0x00000000002a4ab0 0x02b550 0x02b550 R   0x1

 Section to Segment mapping:
  Segment Sections...
   00     .note.gnu.build-id .gnu.hash .dynsym .dynstr .gnu.version .gnu.version_d .gnu.version_r .rela.dyn .rela.plt
   01     .init .plt .plt.got .text .fini
   02     .rodata .eh_frame_hdr .eh_frame
   03     .init_array .fini_array .data.rel.ro .dynamic .got .data .bss
   04     .dynamic
   05     .note.gnu.build-id
   06     .eh_frame_hdr
   07
   08     .init_array .fini_array .data.rel.ro .dynamic .got
```

The 0x078000 offset that corresponds to segment 01 needs to be determined in order to find a function's memory address properly. You can see the full calculation [here](https://github.com/pixie-io/pixie/pull/419/files#diff-e26509bdd023bc2adee54cfb2c8c0c039400d90c6c8e7a2a12edf454278d36ecR518-R521)).

Reply:
Nice. Thanks for the explanation. :)